design-system/packages/fields/src/FieldContainer.tsx

```diff
@@ -1,13 +1,19 @@
 /* @jsx jsx */
 
 import { ReactNode } from 'react';
-import { jsx } from '@keystone-ui/core';
+import { jsx, forwardRefWithAs } from '@keystone-ui/core';
 
 type FieldContainerProps = {
   children: ReactNode;
   className?: string;
 };
 
-export const FieldContainer = ({ children, ...props }: FieldContainerProps) => {
-  return <div {...props}>{children}</div>;
-};
+export const FieldContainer = forwardRefWithAs<'div', FieldContainerProps>(
```

Comment:
The second type arg should be `{}`, and FieldContainerProps should be removed since the children and className will be added based on the `as` prop

Reply:
Removed the classname from `FieldContainerProps` but looks like children type isn't inferred from the forwardWithRefAs fn.

Reply:
It should be, though the `children` prop won't always be passed to the FieldContainer component since not all components you could pass to `as` will accept children so the implementation of FieldContainer needs to not handle children specifically.

Reply:
ah cool, reoslved